npm/install.js

```diff
@@ -36,23 +36,60 @@ fs.unlink(path.join(installScriptLocation, 'cmf-portal'), (err) => {
     }
 });
 
-// download respective release zip from github
-console.info(`Current platform / arch: ${process.platform} / ${process.arch}`);
-const pkgUrl = `https://github.com/criticalmanufacturing/portal-sdk/releases/download/#{Version}#/Cmf.CustomerPortal.Sdk.Console-${process.env.npm_package_version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`;// opts.binUrl.replace("{{version}}", opts.version).replace("{{platform}}", PLATFORM_MAPPING[process.platform]).replace("{{arch}}", ARCH_MAPPING[process.arch]);
-console.info(`Getting release archive from ${pkgUrl} into ${path.resolve(installScriptLocation)}`);
-axios.get(pkgUrl, { responseType: 'arraybuffer' })
-     .then(function (response) {
-         // handle success
-         const zip = tmp.tmpNameSync();
-         console.log(`Writing temporary zip file to ${zip}`);
-         fs.writeFileSync(zip, response.data);
-         console.log(`Extracting zip file ${zip} to ${installScriptLocation}`);
-         (new AdmZip(zip)).extractAllTo(installScriptLocation, undefined, true);
-         rimraf.sync(zip)
-     })
-     .catch(function (error) {
-         // handle error
-         console.error(error);
-         console.error(error(`Could not install version ${process.env.npm_package_version} on your platform ${process.platform}/${process.arch}: ${error.message}`));
-         process.exit(1);
-     });
+// Function to construct the package URL
+const constructUrl = (baseUrl, version) => {
+    return `${baseUrl}/releases/download/${version}/Cmf.CustomerPortal.Sdk.Console-${version}.${PLATFORM_MAPPING[process.platform]}-${ARCH_MAPPING[process.arch]}.zip`;
```

Comment:
since we don't have arguments that usually come from a manual input, this shouldn't be an issue but ideally we should use the language constructs to form URLs. This is due to the fact that there are chars there should be escaped to avoid xss attacks and other things alike.
As a best practice we shouldn't concatenate strings to create URLs.

Reply:
This file was out of date because I had to make some logic changes, but I applied some improvements to the URLs, can you review it now?